app/src/main/java/com/arygm/quickfix/model/profile/searchViewModel.kt

```diff
@@ -13,19 +20,25 @@ class WorkerListViewModel(private val repository: ProfileRepositoryFirestore) :
   fun filterWorkerProfiles(
       hourlyRateThreshold: Double? = null,
       fieldOfWork: String? = null,
-      userLat: Double? = null,
-      userLon: Double? = null,
+      location: Location? = null,
       maxDistanceInKm: Double? = null
   ) {
+    val userLat = location?.latitude
+    val userLon = location?.longitude
+
     repository.filterWorkers(
         hourlyRateThreshold,
         fieldOfWork,
+        location,
+        maxDistanceInKm,
         { profiles ->
           if (userLat != null && userLon != null && maxDistanceInKm != null) {
             val filteredProfiles =
                 profiles.filter { profile ->
-                  val workerLat = profile.location?.latitude ?: return@filter false
-                  val workerLon = profile.location.longitude
+                  val location =
+                      profile.location ?: return@filter false // Ensure location is non-null
+                  val workerLat = location.latitude
+                  val workerLon = location.longitude
```

Comment:
good turnaround to keep only the worker that in the circle of the sepicified radius. Great Job !!